src/layout/layout.js

```diff
@@ -170,23 +175,98 @@
     if (evt.keyCode === this.Keycodes_.ESCAPE &&
         this.drawer_.classList.contains(this.CssClasses_.IS_DRAWER_OPEN)) {
       this.toggleDrawer();
+    } else if (evt.keyCode === this.Keycodes_.SPACE && this.drawer_.classList.contains(this.CssClasses_.IS_DRAWER_OPEN)) {
+      evt.preventDefault();
+      // Send mousedown and mouseup to trigger ripple.
+      var e = new MouseEvent('mousedown');
+      evt.target.dispatchEvent(e);
+      e = new MouseEvent('mouseup');
+      evt.target.dispatchEvent(e);
+      // Send click.
+      evt.target.click();
+    } else if (evt.keyCode === this.Keycodes_.HOME && this.drawer_.classList.contains(this.CssClasses_.IS_DRAWER_OPEN)) {
+      this.drawerMenuItems_[0].focus();
+    } else if (evt.keyCode === this.Keycodes_.END && this.drawer_.classList.contains(this.CssClasses_.IS_DRAWER_OPEN)) {
+      this.drawerMenuItems_[this.drawerMenuItems_.length - 1].focus();
     }
   };
 
   /**
+   * Handles a keyboard navigation events on the drawer.
+   *
+   * @param {Event} evt The event that fired.
+   * @private
+   */
+  MaterialLayout.prototype.handleMenuNavKeyboardEvent_ = function(evt) {
+      // Only react when a draw exists, is open and tab is pressed.
+      if ((evt.keyCode === this.Keycodes_.TAB || evt.keyCode === this.Keycodes_.DOWN_ARROW ||
+            evt.keyCode === this.Keycodes_.UP_ARROW) && this.drawer_ &&
+            this.drawer_.classList.contains(this.CssClasses_.IS_DRAWER_OPEN)) {
+        var currentIndex = Array.prototype.slice.call(this.drawerMenuItems_).indexOf(evt.target);
+        // Tab forward and backward through navigation links. Collapse drawer
+        // when user reaches the end of the links.
```

Comment:
The drawer shouldn't collapse when the end is reached. Tabbing should restart at the first position and continue as normal. This is in alignment with a11y recommended behavior for tab trapping.